GRDB/FTS/FTS5.swift

```diff
@@ -74,6 +74,48 @@ public struct FTS5 {
         #endif
     }
     
+    #if GRDBCUSTOMSQLITE || GRDBCIPHER
+    /// Options for trigram tokenizer character matching. Matches the raw
+    /// "case_sensitive" and "remove_diacritics" tokenizer arguments.
+    ///
+    /// Related SQLite documentation: <https://sqlite.org/fts5.html#the_trigram_tokenizer>
+    public enum TrigramTokenizerMatching: Sendable {
+        /// Case insensitive matching without removing diacritics. This
+        /// option matches the raw "case_sensitive=0 remove_diacritics=0"
+        /// tokenizer argument.
+        case caseInsensitive
+        /// Case insensitive matching that removes diacritics before
+        /// matching. This option matches the raw
+        /// "case_sensitive=0 remove_diacritics=1" tokenizer argument.
+        case caseInsensitiveRemovingDiacritics
+        /// Case sensitive matching. Diacritics are not removed when
+        /// performing case sensitive matching. This option matches the raw
+        /// "case_sensitive=1 remove_diacritics=0" tokenizer argument.
+        case caseSensitive
+    }
```

Comment:
Re-reading myself, I start to wonder if `TrigramCaseSensitiveOption` and `TrigramDiacriticsOption` are that useful.

```swift
t.tokenizer = trigram()
t.tokenizer = trigram(caseSensitive: 1)
t.tokenizer = trigram(caseSensitive: 0, removeDiacritics: 1)
```

This would pass my review as well.

I leave it up to your good taste 😄

Reply:
Thank you for the detailed feedback!

When using the raw `Int?` arguments we obviously can't attach any availability information. Do you think it would make sense to note the SQLite >= 3.45 requirement in the documentation for the `removeDiacritics` parameter? Or better to just leave that to the SQLite documentation entirely since we can't enforce it anyway?

(In theory we could also have two `trigram()` methods as well, one only taking the `caseSensitive` parameter and a second one with both parameters with more restrictive availability. But I don't think that makes sense since that approach won't work if SQLite adds more options for the existing parameters so probably better not to try to enforce availability for the parameters at all than try to partially do it.)

Reply:
> But I don't think that makes sense since that approach won't work if SQLite adds more options for the existing parameters so probably better not to try to enforce availability for the parameters at all than try to partially do it.

Ah. I see. Yes it wouldn't be great if people could write `t.trigram(caseSensitive: 3)` in an app that targets iOS 17. OK, I have been a poor guide here.

So are we back to `TrigramCaseSensitiveOption` and `TrigramDiacriticsOption`, with built-in values subject to availability checkings, along with raw initializers for people who know what they are doing?

Reply:
I think that makes sense, that makes it possible to add availability checks while allowing the user to use raw values to bypass restrictions if needed.

A couple of minor questions:

* You've used optional arguments to configure the options when creating the tokenizer as opposed to using a non-optional default value matching the SQLite default. Do I understand correctly that the idea here is for GRDB to respect the SQLite default, even if that default were to change in the future?
```swift
public static func trigram(
        caseSensitive: FTS5.TrigramCaseSensitiveOption? = nil,
        diacritics: FTS5.TrigramDiacriticsOption? = nil,
    ) -> FTS5TokenizerDescriptor { ... }
```

* Do we make `TrigramDiacriticsOption` itself unavailable for non-custom SQLite or only the `remove` option? I don't think there is a use case for manually using a raw value on OSs where it isn't supported anyway?

* Thoughts on adding `Equatable`, `Hashable` and `BitwiseCopyable` conformances for `TrigramCaseSensitiveOption` and `TrigramDiacriticsOption`; I don't imagine these would see much use but probably no downside to adding them either?